flink-connector-kudu/src/test/java/org/apache/flink/connector/kudu/source/enumerator/KuduSourceEnumeratorStateSerializerTest.java

```diff
@@ -0,0 +1,101 @@
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+/** Tests for {@link KuduSourceEnumeratorStateSerializer}. */
+public class KuduSourceEnumeratorStateSerializerTest {
+    private final KuduSourceEnumeratorStateSerializer serializer =
+            new KuduSourceEnumeratorStateSerializer();
+
+    @Test
+    public void testSerializeDeserialize() throws IOException {
+        byte[] token1 = {1, 2, 3};
+        byte[] token2 = {4, 5, 6};
+        byte[] token3 = {7, 8, 9};
+        byte[] token4 = {10, 11, 12};
+
+        List<KuduSourceSplit> unassigned =
+                Arrays.asList(new KuduSourceSplit(token1), new KuduSourceSplit(token2));
+        List<KuduSourceSplit> pending =
+                Arrays.asList(new KuduSourceSplit(token3), new KuduSourceSplit(token4));
+        KuduSourceEnumeratorState state =
+                new KuduSourceEnumeratorState(12345L, unassigned, pending);
+
+        byte[] serialized = serializer.serialize(state);
+        KuduSourceEnumeratorState deserialized =
+                serializer.deserialize(serializer.getVersion(), serialized);
+
+        assertThat(state.getLastEndTimestamp()).isEqualTo(deserialized.getLastEndTimestamp());
+        assertThat(state.getUnassigned().size()).isEqualTo(deserialized.getUnassigned().size());
+        assertThat(state.getPending().size()).isEqualTo(deserialized.getPending().size());
+
+        for (KuduSourceSplit split : unassigned) {
+            assertThat(split.getSerializedScanToken())
+                    .isEqualTo(
+                            deserialized.getUnassigned().stream()
+                                    .filter(
+                                            dSplit ->
+                                                    Arrays.equals(
+                                                            dSplit.getSerializedScanToken(),
+                                                            split.getSerializedScanToken()))
+                                    .findFirst()
+                                    .orElseThrow(() -> new AssertionError("Missing expected split"))
+                                    .getSerializedScanToken());
+        }
+
+        for (KuduSourceSplit split : pending) {
+            assertThat(split.getSerializedScanToken())
+                    .isEqualTo(
+                            deserialized.getPending().stream()
+                                    .filter(
+                                            dSplit ->
+                                                    Arrays.equals(
+                                                            dSplit.getSerializedScanToken(),
+                                                            split.getSerializedScanToken()))
+                                    .findFirst()
+                                    .orElseThrow(() -> new AssertionError("Missing expected split"))
+                                    .getSerializedScanToken());
+        }
```

Comment:
This could be simplified if we use a traditional `for` loop:
```java
for (int i = 0; i < unassigned.size(); i++) {
    byte[] expected = unassigned.get(i).getSerializedScanToken();
    assertThat(deserialized.getUnassigned().get(i).getSerializedScanToken())
            .isEqualTo(expected);
}

for (int i = 0; i < pending.size(); i++) {
    byte[] expected = pending.get(i).getSerializedScanToken();
    assertThat(deserialized.getPending().get(i).getSerializedScanToken())
            .isEqualTo(expected);
}
```